src/wasm/wasm.cpp

```diff
@@ -760,7 +760,14 @@ void Binary::finalize() {
   }
 }
 
-void Select::finalize(Type type_) { type = type_; }
+void Select::finalize(Type type_) {
+  assert(ifTrue && ifFalse);
+  if (ifTrue->type == Type::unreachable || ifFalse->type == Type::unreachable) {
```

Comment:
I think this also include the condition?

Reply:
Oh right 😅